sdk/core/azure-core-amqp/docs/architecture.md

```diff
@@ -0,0 +1,53 @@
+# Architecture Docs
+
+## Qpid Proton-J Integration
+
+Qpid Proton-J publishes events and messages via its event-driven process called [`Reactor`][Reactor]. `azure-core-amqp`
+hooks into Qpid Proton-J Reactor via `Handlers`.  [`BaseHandler`][BaseHandler] contains all of the events that can be
```

Comment:
Is it the case that once we create a `Reactor` object then that object creates and owns a `BaseHandler` object, and we can register our own more granular user-handler objects? Like one for connection-events, another for session-events etc.. Where base-handler demultiplex events to these specialized user-handlers?